### PR TITLE
Update winstone to 5.24

### PR DESCRIPTION
**Pull request by Vlatombe**
* Downstream of https://github.com/jenkinsci/winstone/pull/174

See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

### Proposed changelog entries

* Winstone 5.24 - Add an option to write the listening port to a file [changelog](https://github.com/jenkinsci/winstone/releases/tag/winstone-5.24), Remove automatic self signed cert if tls specified but no keystore [changelog](https://github.com/jenkinsci/winstone/releases/tag/winstone-5.23)

### Proposed upgrade guidelines

In previous versions of Jenkins, a self-signed certificate was used if no HTTPS keystore was specified. This feature has been removed.

If using HTTPS or HTTP/2, you will need to make sure an HTTPS keystore is configured by adding the `--httpsKeyStore=path/to/keystore` and `--httpsKeyStorePassword=keystorePassword` arguments to Winstone.

See https://www.jenkins.io/doc/book/installing/initial-settings/#https-with-an-existing-certificate & https://www.jenkins.io/doc/book/installing/initial-settings/#https-certificates-with-windows for more info on creating a key store.

### Submitter checklist

- [X] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by jglick** [comment]
@Vlatombe is this still active or can it be closed?

File: war/pom.xml
Comment (Vlatombe) [inline war/pom.xml:112]:
```suggestion
      <version>5.24</version>
```

**Review by timja — APPROVED** [review]

**Review by jglick — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Comment by basil** [comment]
I have added the `upgrade-guide-needed` label to this PR based on the fact that it caused a user to file [JENKINS-68107](https://issues.jenkins.io/browse/JENKINS-68107).

@dskrvk Since you were so kind as to provide https://github.com/jenkins-infra/jenkins.io/pull/5021 would you also be willing to write a small blurb for the upgrade guide? No pressure, though. If, on the other hand, that is asking too much, I can try to write one up myself.

**Comment by dskrvk** [comment]
How about this (LMK if these notes are kept in source control somewhere, and I should make a PR):
```
If using HTTP/2, you'll need to make sure an HTTPS key store is configured (even if no HTTPS port is set). Depending on how you run Jenkins this can be either:
1. provided through the `java` command line args (right after `--http2Port=XXXX`): `--httpsKeyStore=path/to/keystore --httpsKeyStorePassword=keystorePassword`, or
2. specified in the system service config file, e.g. on CentOS-like systems this is `/etc/sysconfig/jenkins`, where you'll need to populate `JENKINS_HTTPS_KEYSTORE` and `JENKINS_HTTPS_KEYSTORE_PASSWORD`.

See https://www.jenkins.io/doc/book/installing/initial-settings/#https-with-an-existing-certificate & https://www.jenkins.io/doc/book/installing/initial-settings/#https-certificates-with-windows for more info on creating a key store.
```

**Comment by basil** [comment]
Thanks @dskrvk! Our process is to just leave a note in the original PR, and the members of the docs team will extract that and include it in the upgrade guide for the next LTS. Accordingly, I have updated the description for this PR to include a (slightly edited) version of the note you provided.

**Pull request merged**
